src/wazuh_modules/task_manager/wm_task_manager.c

```diff
@@ -203,66 +222,203 @@ cJSON* wm_task_manager_analyze_task(const cJSON *task_object, int *error_code) {
+            *error_code = WM_TASK_INVALID_AGENT_ID;
+            response = wm_task_manager_build_response(WM_TASK_INVALID_AGENT_ID, agent_id, task_id, status);
         }
 
     } else {
-        *error_code = WM_TASK_INVALID_MODULE;
-        response = wm_task_manager_build_response(WM_TASK_INVALID_MODULE, agent_id, task_id, status);
+        *error_code = WM_TASK_INVALID_COMMAND;
+        response = wm_task_manager_build_response(WM_TASK_INVALID_COMMAND, agent_id, task_id, status);
     }
 
-    os_free(status);
+    if (status){
+        os_free(status);
+    }
+    return response;
+}
+
+cJSON* wm_task_manager_analyze_task_api_module(const cJSON *task_object, int *error_code) {
+    cJSON *tmp = NULL;
+    cJSON *response = NULL;
+    int result = 0;
+    int agent_id = OS_INVALID;
+    int task_id = OS_INVALID;
+    char *status = NULL;
+    int create_time = OS_INVALID;
+    int last_update_time = OS_INVALID;
+    char *command_result = NULL;
+    char *module_result = NULL;
+
+    char *command = cJSON_GetObjectItem(task_object, json_keys[WM_TASK_COMMAND])->valuestring;
+
+    if (tmp = cJSON_GetObjectItem(task_object, json_keys[WM_TASK_AGENT_ID]), tmp && tmp->type == cJSON_Number) {
+        agent_id = tmp->valueint;
+    }
+    if (tmp = cJSON_GetObjectItem(task_object, json_keys[WM_TASK_TASK_ID]), tmp && tmp->type == cJSON_Number) {
+        task_id = tmp->valueint;
+    }
+    if (tmp = cJSON_GetObjectItem(task_object, json_keys[WM_TASK_STATUS]), tmp && tmp->type == cJSON_String) {
+        os_strdup(tmp->valuestring, status);
+    }
+
+    if (!strcmp(commands_list[WM_TASK_UPGRADE_RESULT], command)) {
+            
+        if (agent_id != OS_INVALID) {
+            if (task_id = wm_task_manager_get_task_by_agent_id_and_module(agent_id, modules_list[WM_TASK_UPGRADE_MODULE], &command_result, &status, &create_time, &last_update_time), task_id == OS_INVALID){
+                *error_code = WM_TASK_DATABASE_ERROR;
+                response = wm_task_manager_build_response(WM_TASK_INVALID_AGENT_ID, agent_id, task_id, status);
+            } else if (task_id == OS_NOTFOUND || task_id == 0) {
+                *error_code = WM_TASK_DATABASE_NO_TASK;
+                response = wm_task_manager_build_response(WM_TASK_DATABASE_NO_TASK, agent_id, OS_INVALID, status);
+            } else {
+                response = wm_task_manager_build_response(WM_TASK_SUCCESS, agent_id, task_id, NULL);
+                wm_task_manager_build_response_result(response, modules_list[WM_TASK_UPGRADE_MODULE], command_result, status, create_time, last_update_time, command);
+                os_free(command_result);
+            }
+        } else {
+            *error_code = WM_TASK_INVALID_AGENT_ID;
+            response = wm_task_manager_build_response(WM_TASK_INVALID_AGENT_ID, agent_id, task_id, status);
+        }
+    } else if (!strcmp(commands_list[WM_TASK_TASK_RESULT], command)) {
+
```

Comment:
First check if `task_id` is not `OS_INVALID`:
```c
if (task_id != OS_INVALID) {
...
} else {
    *error_code = WM_TASK_INVALID_TASK_ID;
    response = wm_task_manager_build_response(WM_TASK_INVALID_TASK_ID, agent_id, task_id, status);
}
```

Reply:
done